### PR TITLE
HCM Admin Console v0.3  & Microplan v0.1  Release UI Code changes

### PR DESCRIPTION
**Pull request by jagankumar-egov**
## Choose the appropriate template for your PR:

- [Feature PR](./PULL_REQUEST_TEMPLATE/feature_template.md)
- [Bugfix PR](./PULL_REQUEST_TEMPLATE/bugfix_template.md)
- [Release PR](./PULL_REQUEST_TEMPLATE/release_template.md)

## Release Notes

- **New Features**
	- Introduced a comprehensive search functionality for checklists and campaigns.
	- Added new components for managing boundary relationships and microplan data.
	- Enhanced UI with new styles and improved user interactions across various components.
	- Implemented a new loading state management system for better user experience during data fetching.
	- Added a new configuration for microplan campaign search functionality.
	- Introduced a new React component for managing approved microplans.
	- Added functionality for creating and managing boundary relations.
	- Introduced new components for campaign data fetching and management, enhancing user experience.

- **Bug Fixes**
	- Resolved issues with data handling and state management in several components.
	- Improved error handling and notifications for user actions.

- **Documentation**
	- Updated changelogs for the campaign manager and web modules to reflect recent changes.
	- Added a new README and detailed comments in the code for better understanding.

- **Style**
	- Refined CSS styles for improved layout and responsiveness across the application.
	- Introduced new SCSS files for specific components to maintain modularity.

- **Tests**
	- Enhanced test coverage for new features and critical components to ensure reliability.

### CODE REVIEW

**Review by nabeelmd-eGov — DISMISSED** [review]

File: health/micro-ui/web/micro-ui-internals/packages/modules/campaign-manager/src/components/SelectingBoundaryComponent.js [hunk 1]
```diff
@@ -0,0 +1,655 @@
...
+//     code: "Nampula.Murrupula",
+//     name: "HEALTH_MO_13_01_MURRUPULA",
+//     boundaryType: "District",
+//   },
+//   {
+//     code: "Cabo.Cabo Delgado",
+//     name: "HEALTH_MO_12_01_CABO DELGADO",
+//     boundaryType: "District",
+//   },
+//   {
```
[the remark this answers is not in the document]
Comment (Bhavya-egov) [reply health/micro-ui/web/micro-ui-internals/packages/modules/campaign-manager/src/components/SelectingBoundaryComponent.js:43]: It is the sample data for the frozen data.

File: health/micro-ui/web/micro-ui-internals/packages/modules/campaign-manager/src/components/BoundaryDetailsSummary.js [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+  //         ))}
+
+  //       {/* Show the button if there are more than 2 parent keys */}
+  //       {Object.keys(groupedByParent).length > 2 && (
+  //         <Button label={showAll ? t("HCM_SHOW_LESS_ALL") : t("HCM_SHOW_MORE_ALL")} onClick={handleShowAllToggle} variation="link" />
+  //       )}
+  //     </div>
+  //   </div>
+  // );
+
```
[the remark this answers is not in the document]
Comment (Bhavya-egov) [reply health/micro-ui/web/micro-ui-internals/packages/modules/campaign-manager/src/components/BoundaryDetailsSummary.js:85]: done

File: health/micro-ui/web/micro-ui-internals/packages/modules/campaign-manager/src/components/CampaignSummary.js [hunk 3]
```diff
@@ -162,39 +183,198 @@ const CampaignSummary = (props) => {
...
+  //   //   setTargetErrors(props?.props?.summaryErrors?.targetErrors);
+  //   // }
+  //   // if (props?.props?.summaryErrors?.facilityErrors) {
+  //   //   setFacilityErrors(props?.props?.summaryErrors?.facilityErrors);
+  //   // }
+  //   // if (props?.props?.summaryErrors?.userErrors) {
+  //   //   setUserErrors(props?.props?.summaryErrors?.userErrors);
+  //   // }
+  // }, [props?.props?.summaryErrors]);
+
```
[the remark this answers is not in the document]
Comment (Bhavya-egov) [reply health/micro-ui/web/micro-ui-internals/packages/modules/campaign-manager/src/components/CampaignSummary.js:359]: done

File: health/micro-ui/web/micro-ui-internals/packages/modules/campaign-manager/src/components/CampaignSummary.js [hunk 4]
```diff
@@ -209,45 +389,56 @@ const CampaignSummary = (props) => {
...
+            //             component: "TimelineComponent",
+            //             props: {
+            //               campaignId: data?.[0]?.id,
+            //               resourceId: resourceIdArr,
+            //             },
+            //           },
+            //         ]
+            //       : []),
+            //   ],
+            // },
```
[the remark this answers is not in the document]
Comment (Bhavya-egov) [reply health/micro-ui/web/micro-ui-internals/packages/modules/campaign-manager/src/components/CampaignSummary.js:421]: done

**Review by nabeelmd-eGov — APPROVED** [review]
Looks fine to me

**Review by nipunarora-eGov — COMMENTED** [review]
Based on High level review, these are the points/ tech debts I can thought of:

1. Consistency in localization module names → some having (prefix) → It should be configurable if required
2. UI master for Pop and Plan Inboxes
3. Using common components wherever customizations have been made
4. Enhance the multiselect dropdown component → use the same from UI-components in boundary screens
5. Re-design/Cleanup of assumption and formula flow as the current code has gotten very complex and unreadable for a new dev
6. Using Redux for global states instead of context API
7. In many screens duplicate API calls are made due to multiple unnecessary re-renders and state changes which trigger API calls. We need to fix this 
8. Writing a wrapper over react-data-table library's Table component and publishing as part of ui-components package
9. Reworking plan and pop inboxes with help of UI masters(stuffs like button icons, order of action buttons, some validations logics are hard coded) we can extract those to in mdms masters
10. We should to keep a CONSTANTS file in UI so we can import that and use instead of hardcoding actions, user role codes etc....

**Review by nipunarora-eGov — APPROVED** [review]
Added my observations/tech debts that I could think of at the moment.

**Review by nipunarora-eGov — APPROVED** [review]

**Review by nabeelmd-eGov — APPROVED** [review]

**Pull request merged**